economy/models.py

```diff
@@ -152,31 +152,46 @@ class ProductOrder(models.Model):
     )
 
     order_size = models.IntegerField(default=1)
-    amount = models.IntegerField()
 
-    purchase = models.ForeignKey(
-        'Purchase',
+    source = models.ForeignKey(
+        'SociBankAccount',
         related_name='product_orders',
-        on_delete=models.CASCADE
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL
     )
 
+    session = models.ForeignKey(
+        'SociSession',
+        related_name='product_orders',
+        on_delete=models.DO_NOTHING,
+        default=SociSession.get_active_session
+    )
+
+    @property
+    def total_sum(self):
+        return self.product.price * self.order_size
+
     def __str__(self):
-        return f"Order of {self.order_size} {self.product.name}(s)"
+        return f"Order of {self.order_size} {self.product.name}(s) by " \
+               f"{self.source.user} of {self.total_sum}kr"
 
     def __repr__(self):
         return f"Order(product={self.product}, order_size={self.order_size})"
 
     def save(self, force_insert=False, force_update=False, using=None, update_fields=None):
         if self._state.adding:
-            order_amount = self.amount * self.order_size
-            self.purchase.source.remove_funds(amount=order_amount)
-            SociBankAccount.soci_master_account.get().add_funds(amount=order_amount)
+            if self.session is None:
+                raise NoSociSessionError()
+            self.source.remove_funds(amount=self.total_sum)
+            SociBankAccount.soci_master_account.get().add_funds(amount=self.total_sum)
 
         super().save(force_insert=force_insert, force_update=force_update, using=using, update_fields=update_fields)
 
 
 class Purchase(TimeStampedModel):
     """
+    ToBe deprecated
```

Comment:
The reason I think we need both `ProductOrder` and `Purchase` is because it should be possible order multiple quantities of multiple products in a single "kryss". If you remove `Purchase`, you lose the ability to track which `ProductOrders` the user made at the same time. You can of course aggregate this data, but it would not be as precise as using a `Purchase` model.

Reply:
This was mostly due to the limitations in the frontend implementation. There was no straightforward way to implement a threestage forgeign key relationship without complex and cumbersome form handling. Seeing as you wrote these models to begin with if you have a proposition for how to refactor the current implementation to accommodate the Purchase model I would be happy to take another gander at it.